lib/auth/password_test.go

```diff
@@ -117,7 +117,87 @@ func setupPasswordSuite(t *testing.T) *passwordSuite {
 	return &s
 }
 
-func TestUserNotFound(t *testing.T) {
+func TestCheckPassswordWOToken(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	s := setupPasswordSuite(t)
+
+	u, err := types.NewUser("user1")
+	require.NoError(t, err)
+	_, err = s.a.CreateUser(ctx, u)
+	require.NoError(t, err)
+	require.NoError(t, s.a.UpsertPassword("user1", []byte("supersecret123")))
+
+	assert.NoError(t, s.a.checkPasswordWOToken("user1", []byte("supersecret123")))
+
+	err = s.a.checkPasswordWOToken("user1", []byte("wrongpassword"))
+	assert.Error(t, err)
+	assert.True(t, trace.IsBadParameter(err))
+}
+
+func TestCheckPassswordWOToken_legacyUser(t *testing.T) {
```

Comment:
Nit: As someone whose not too familiar with what you are working on - "legacy user" doesn't mean much to me. Perhaps a short comment would help provide some context.

Reply:
Made the function name more descriptive.